src/ember-app.js

```diff
@@ -235,7 +239,7 @@ class EmberApp {
       })
       .then(() => result.instanceBooted = true)
       .then(() => instance.visit(path, bootOptions))
-      .then(() => waitForApp(instance))
+      .then(() => fastbootInfo.deferredPromise)
```

Comment:
even if `fastbootInfo` is mutated it points to same object right, and it is registered in registry as `{ instantiate: false }` which means registry doesn't do anything special it comes out same as it comes in, correct if I am wrong :)